jc/parsers/net_user.py

```diff
@@ -0,0 +1,441 @@
+                    "Administrators"
+                ],
+                "logon_hours_allowed": "All",
+                "password_changeable": "2021-12-17T11:07:14",
+                "password_expires": "2022-01-27T11:07:14",
+                "password_last_set": "2021-12-16T11:07:14",
+                "password_required": true,
+                "user_may_change_password": true,
+                "user_name": "Administrators",
+                "workstations_allowed": "All"
+            }
+        ]
+    }
+
+    $ net users Administrator /domain | jc --net-user -p | jq
+    {
+        "domain": "somecompany.corp",
+        "user_accounts": [
+            {
+                "account_active": true,
+                "account_expires": "Never",
+                "comment": "Built-in account for administering the computer/domain",
+                "country_region_code": "000 (System Default)",
+                "global_group_memberships": [
+                    "Domain Admins",
+                    "Domain Users",
+                    "Group Policy Creator",
+                    "Enterprise Admins",
+                    "Schema Admins"
+                ],
+                "last_logon": "2024-07-17T13:46:12",
+                "local_group_memberships": [
+                    "Administrators"
+                ],
+                "logon_hours_allowed": "All",
+                "password_changeable": "2023-09-30T11:44:26",
+                "password_expires": "Never",
+                "password_last_set": "2023-09-29T11:44:26",
+                "password_required": true,
+                "user_may_change_password": true,
+                "user_name": "Administrators",
+                "workstations_allowed": "All"
+            }
+        ]
+    }
+"""
+
+
+from datetime import datetime
+import re
+import jc.utils
+
+
+class info():
+    """Provides parser metadata (version, author, etc.)"""
+    version = '1.0'
+    description = '`net user` command parser'
+    author = 'joehacksalot'
+    author_email = 'joehacksalot@gmail.com'
+    compatible = ['windows']
+    magic_commands = ['net-user']
+    tags = ['command']
+
+
+__version__ = info.version
+
+
+def parse(data, raw=False, quiet=False):
+    """
+    Main text parsing function
+
+    Parameters:
+
+        data:        (string)  text data to parse
+        raw:         (boolean) unprocessed output if True
+        quiet:       (boolean) suppress warning messages if True
+
+    Returns:
+
+        Parsed dictionary. The raw and processed data structures are the same.
+    """
+    jc.utils.compatibility(__name__, info.compatible, quiet)
+    jc.utils.input_type_check(data)
+
+    raw_output = {}
+    if jc.utils.has_data(data):
+        try:
+            raw_output = _parse(data)
+            return raw_output if raw else _process(raw_output)
+        except Exception as e:
+            if not quiet:
+                jc.utils.warning_message(['Could not parse data due to unexpected format.'])
```

Comment:
This type of exception handling can make it difficult to troubleshoot. Doing this means the user cannot use `jc --route-print -dd` to get debug info. Without the exception handling `jc` should just do the right thing. If you want a custom exception message, it would be better to use `jc.exceptions.ParseError`:

```
from jc.exceptions import ParseError
raise ParseError('My custom error here.')
```
You can find examples in the following parsers: cef.py, csv.py, uname.py, etc.